### PR TITLE
Ignore stubgen errors and wrap TriangulationFactor

### LINKED ISSUE #1758: TriangulationFactor Python
Hi, I can't seem to find TriangulationFactor in the python bindings? Any idea why it isn't exported?

### LINKED ISSUE #1791: Boostless GTSAM built requires Boost as dependency
## Description

I built GTSAM without boost, yet linking it in my application leads to CMake looking for Boost. This is the CMake error:
```c
CMake Error at C:/Program Files/CMake/share/cmake-3.30/Modules/FindPackageHandleStandardArgs.cmake:233 (message):
  Could NOT find Boost (missing: Boost_INCLUDE_DIR)
```

 This is the GTSAMConfig.cmake:

```c
# - Config file for GTSAM
# It defines the following variables
#  GTSAM_INCLUDE_DIR - include directories for GTSAM

# Compute paths
get_filename_component(OUR_CMAKE_DIR "${CMAKE_CURRENT_LIST_FILE}" PATH)
if(EXISTS "${OUR_CMAKE_DIR}/CMakeCache.txt")
  # In build tree
  set(GTSAM_INCLUDE_DIR C:/Users/cipadmin/Documents/Workspace/spear/3rd/gtsam CACHE PATH "GTSAM include directory")
else()
  # Find installed library
  set(GTSAM_INCLUDE_DIR "${OUR_CMAKE_DIR}/../include" CACHE PATH "GTSAM include directory")
endif()

# Find dependencies, required by cmake exported targets:
include(CMakeFindDependencyMacro)
# Allow using cmake < 3.8
if(${CMAKE_VERSION} VERSION_LESS "3.8.0")
find_package(Boost  COMPONENTS )
else()
find_dependency(Boost  COMPONENTS )
endif()

if(0)
    find_dependency(TBB 4.4 COMPONENTS tbb tbbmalloc)
endif()

if(OFF)
find_dependency(Eigen3 REQUIRED)
endif()

# Load exports
include(${OUR_CMAKE_DIR}/GTSAM-exports.cmake)

# Load project-specific flags, if present
if(EXISTS "${OUR_CMAKE_DIR}/gtsam_extra.cmake")
	include("${OUR_CMAKE_DIR}/gtsam_extra.cmake")
endif()

message(STATUS "GTSAM include directory:  ${GTSAM_INCLUDE_DIR}")
```

## Steps to reproduce

1.  Build GTSAM with the following commands:
```c
cmake .. -DGTSAM_USE_BOOST_FEATURES=OFF -DGTSAM_ENABLE_BOOST_SERIALIZATION=OFF -DCMAKE_INSTALL_PREFIX=../cmake-install/ -DCMAKE_BUILD_TYPE=Release  -DGTSAM_BUILD_UNSTABLE=OFF -DGTSAM_BUILD_PYTHON=OFF
```

2. Link GTSAM in another CMake project

## Expected behavior

It should not link to Boost if it is built without Boost.

## Environment

Windows 10

### PR DESCRIPTION
**Pull request by varunagrawal**
Until #1824 is addressed, I suggest we keep using mypy's stubgen.
I've kept the `#include<algorithm>` for compilation.

Additionally, fixes #1758.

Updated to also address #1791.

### CODE REVIEW

**Comment by ProfFan** [comment]
I think it does not matter? Mypy has the same issue, they just pretend that it does not exist, the generated signature is `Any`, which `pybind11-stubgen` also do anyways. #1824 is universal to any generator.

File: cmake/Config.cmake.in [hunk 1]
```diff
@@ -15,10 +15,12 @@ endif()
 # Find dependencies, required by cmake exported targets:
 include(CMakeFindDependencyMacro)
 # Allow using cmake < 3.8
-if(${CMAKE_VERSION} VERSION_LESS "3.8.0")
-find_package(Boost @BOOST_FIND_MINIMUM_VERSION@ COMPONENTS @BOOST_FIND_MINIMUM_COMPONENTS@)
-else()
-find_dependency(Boost @BOOST_FIND_MINIMUM_VERSION@ COMPONENTS @BOOST_FIND_MINIMUM_COMPONENTS@)
+if (@GTSAM_ENABLE_BOOST_SERIALIZATION@ OR @GTSAM_USE_BOOST_FEATURES@)
```
Comment (ProfFan) [inline cmake/Config.cmake.in:18]: How does this fix #1841 ?

Comment (varunagrawal) [reply cmake/Config.cmake.in:18]: I don't claim it does. It fixes #1791.

**Comment by varunagrawal** [comment]
> I think it does not matter? Mypy has the same issue, they just pretend that it does not exist, the generated signature is `Any`, which `pybind11-stubgen` also do anyways. #1824 is universal to any generator.

Yes but at least we don't get bombarded by a bunch of error messages. Those error messages are very misleading and for someone not familiar with the wrapper and stub generation, it can be very confusing.

**Comment by ProfFan** [comment]
> > I think it does not matter? Mypy has the same issue, they just pretend that it does not exist, the generated signature is `Any`, which `pybind11-stubgen` also do anyways. #1824 is universal to any generator.
> 
> Yes but at least we don't get bombarded by a bunch of error messages. Those error messages are very misleading and for someone not familiar with the wrapper and stub generation, it can be very confusing.

I see, if that's the use case the only change needed is `--ignore-all-errors` in the command line

File: python/CMakeLists.txt [hunk 2]
```diff
@@ -251,22 +250,6 @@ if(GTSAM_UNSTABLE_BUILD_PYTHON)
             VERBATIM            
         )
     endif()
-
-    add_custom_target(
-        python-unstable-stubs
-        COMMAND
-        ${CMAKE_COMMAND} -E env
-        "PYTHONPATH=${GTSAM_PYTHON_BUILD_DIRECTORY}/$ENV{PYTHONPATH}"
-        pybind11-stubgen -o . --enum-class-locations \"KernelFunctionType|NoiseFormat:gtsam.gtsam\" --enum-class-locations \"OrderingType:gtsam.gtsam.Ordering\" --numpy-array-use-type-var gtsam_unstable
```
Comment (ProfFan) [inline python/CMakeLists.txt:260]: Here: `pybind11-stubgen -o . --enum-class-locations \"KernelFunctionType|NoiseFormat:gtsam.gtsam\" --enum-class-locations \"OrderingType:gtsam.gtsam.Ordering\" --numpy-array-use-type-var --ignore-all-errors gtsam_unstable`

File: cmake/Config.cmake.in (hunk 1 above)
Comment (ProfFan) [reply cmake/Config.cmake.in:18]: My bad

File: python/CMakeLists.txt (hunk 2 above)
Comment (varunagrawal) [reply python/CMakeLists.txt:260]: Thanks! Updated

**Review by ProfFan — APPROVED** [review]
LGTM :)

**Pull request merged**